bldsys/cmake/component_helper.cmake

```diff
@@ -0,0 +1,163 @@
+    set(TARGET_FOLDER "components/${TARGET_NAME}")
+
+    set(LINKAGE "UNKNOWN") # Used to determine whether to create a static or interface library
+
+    if(TARGET_SRC)
+        message(STATUS "STATIC: ${TARGET}")
+        set(LINKAGE "PUBLIC")
+        add_library(${TARGET} STATIC ${TARGET_SRC} ${TARGET_HEADERS})
+
+        if(${VKB_WARNINGS_AS_ERRORS})
+            if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang" OR "${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
+                target_compile_options(${TARGET} PRIVATE -Werror)
+                # target_compile_options(${TARGET} PRIVATE -Wall -Wextra -Wpedantic -Werror)
+            elseif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "MSVC")
+                target_compile_options(${TARGET} PRIVATE /W3 /WX)
+                # target_compile_options(${TARGET} PRIVATE /W4 /WX)
+            endif()
+        endif()
+    else()
+        message("INTERFACE: ${TARGET}")
+        set(LINKAGE "INTERFACE")
+        add_library(${TARGET} INTERFACE)
+        target_sources(${TARGET} INTERFACE ${TARGET_HEADERS})
+    endif()
+
+    if (LINKAGE STREQUAL "UNKNOWN")
+        message(FATAL_ERROR "Failed to determine linkage type for ${TARGET}")
+    endif()
+
+    if(TARGET_LINK_LIBS)
+        target_link_libraries(${TARGET} ${LINKAGE} ${TARGET_LINK_LIBS})
+    endif()
+
+    if(TARGET_INCLUDE_DIRS)
+        target_include_directories(${TARGET} ${LINKAGE} ${TARGET_INCLUDE_DIRS})
+    endif()
+
+    target_compile_features(${TARGET} ${LINKAGE} cxx_std_17)
+
+    set_property(TARGET ${TARGET} PROPERTY FOLDER ${TARGET_FOLDER})
+
+    if(NOT TARGET_NO_DEFAULT_INCLUDES)
+        target_include_directories(${TARGET} ${LINKAGE} ${CMAKE_CURRENT_SOURCE_DIR}/include)
+    endif()
+
+    add_dependencies(vkb__components ${TARGET})
+endfunction()
+
+# Enable testing within the components
+macro(vkb__enable_testing)
+    if((CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME AND BUILD_TESTING) OR VKB_BUILD_TESTS)
+        include(CTest)
+        enable_testing()
+    endif()
+endmacro()
+
+# Bucket target for all tests
+add_custom_target(vkb__tests)
+
+# Register a new test
+# Adds the test to the vkb__tests target
+# Uses Catch2 by default. If NO_CATCH2 is set, then the test must provide its own main function
+function(vkb__register_tests)
+    set(options NO_CATCH2)
+    set(oneValueArgs COMPONENT NAME)
+    set(multiValueArgs SRC HEADERS LINK_LIBS INCLUDE_DIRS)
+
+    if(NOT((CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME AND BUILD_TESTING) OR VKB_BUILD_TESTS))
+        return() # testing not enabled
+    endif()
+
+    cmake_parse_arguments(TARGET "${options}" "${oneValueArgs}" "${multiValueArgs}" ${ARGN})
+
+    if(TARGET_NAME STREQUAL "")
+        message(FATAL_ERROR "NAME must be defined in vkb__register_tests")
+    endif()
+
+    if(NOT TARGET_SRC)
+        message(FATAL_ERROR "One or more source files must be added to vkb__register_tests")
+    endif()
+
+    set(TARGET_FOLDER "components/${TARGET_COMPONENT}")
```

Comment:
Maybe set the folder to something like "tests" ?

Reply:
Ah this was so that the test target showed up close to the component target. But I can remove the `${TARGET_COMPONENT}` part so that it matches your proposed changes in the screenshots